accounts-on-demand/src/subscription_manager.rs

```diff
@@ -0,0 +1,281 @@
+                        },
+                    );
+                }
+            }
+
+            let subscribe_request = SubscribeRequest {
+                accounts: subscribe_accounts,
+                slots: Default::default(),
+                transactions: Default::default(),
+                blocks: Default::default(),
+                blocks_meta: Default::default(),
+                entry: Default::default(),
+                commitment: Some(processed_commitment.into()),
+                accounts_data_slice: Default::default(),
+                ping: None,
+            };
+
+            log::info!(
+                "Accounts on demand subscribing to {}",
+                grpc_config.grpc_addr
+            );
+            let mut client = yellowstone_grpc_client::GeyserGrpcClient::connect(
+                grpc_config.grpc_addr.clone(),
+                grpc_config.grpc_x_token.clone(),
+                None,
+            )
+            .unwrap();
+            let mut account_stream = client.subscribe_once2(subscribe_request).await.unwrap();
+
+            while let Some(message) = account_stream.next().await {
+                let message = message.unwrap();
+                let Some(update) = message.update_oneof else {
+                    continue;
+                };
+
+                match update {
+                    UpdateOneof::Account(account) => {
+                        if let Some(account_data) = account.account {
+                            let account_pk_bytes: [u8; 32] = account_data
+                                .pubkey
+                                .try_into()
+                                .expect("Pubkey should be 32 byte long");
+                            let owner: [u8; 32] = account_data
+                                .owner
+                                .try_into()
+                                .expect("owner pubkey should be deserializable");
+                            let notification = AccountNotificationMessage {
+                                data: AccountData {
+                                    pubkey: Pubkey::new_from_array(account_pk_bytes),
+                                    account: Arc::new(Account {
+                                        lamports: account_data.lamports,
+                                        data: account_data.data,
+                                        owner: Pubkey::new_from_array(owner),
+                                        executable: account_data.executable,
+                                        rent_epoch: account_data.rent_epoch,
+                                    }),
+                                    updated_slot: account.slot,
+                                },
+                                // TODO update with processed commitment / check above
+                                commitment: Commitment::Processed,
+                            };
+                            if account_stream_sx.send(notification).is_err() {
```

Comment:
why are you doing that? the error happens if broadcast has no receivers. that's not an error and it does not indicate a problem with upstream. or am I missing something?

Reply:
Nah if there are no subscption it means that there are no subscriptions for accounts for the moment. Should not do anything in that case.

Reply:
> Nah if there are no subscption it means that means no one is listening to the geyser stream.